greenplumpython/table.py

```diff
@@ -52,6 +55,85 @@ def select(self, target_list: Iterable) -> "Table":
             parents=[self],
         )
 
+    def _join(
+        self,
+        other: "Table",
+        targets1: List = ["*"],
+        targets2: List = ["*"],
```

Comment:
> we should have tests on the `_join()` to cover most of the scenarios of different input targets. Since other public methods are simple wrappers of the `_join()`, that would be enough. In the tests cases, we need to test:
> 
> 1. both targets have default values. (`SELECT *, * from t1 join t2 on t1.id = t2.id`, this would have questionable behavior, why does the user ever need two `*` ?)
> 2. both targets have empty targets
> 3. One empty, one default
> 4. both have multiple targets
> 5. non-exist target, do we have a clear error?
> 6. other necessary combinations.
> 
> The tests will focus on the columns, so we don't have to test the tuples to simplify the cases.
> 
> If we are uncertain about some cases, leave a FIXME in the code so we can fix them later.

1.To handle both `*` : in order to manage same name columns, rename columns which have identical name than others in format "TableName_ColumnName'  
2. If both targets lists empty, same as two `*`
5. Raise error if column not exists

Reply:
I am thinking about having something like
```python
t1.join(t2, targets=[t1["id"].rename_to("t1_id"), t2["id"].rename_to("t2_id")])
```
where `rename_to()` is a method of an `Expr`:
```python
def rename_to(self, new_name: str) -> Expr:
    new_expr = copy.copy(self) # Shallow copy
    new_expr._as_name = new_name
    return new_expr
```